neuron/src/app/Neuron/Config.hs

```diff
@@ -35,23 +37,43 @@ import System.FilePath
 getConfig :: Action Config
 getConfig = do
   configPath <- ribInputDir <&> (</> configFile)
-  configVal :: Text <-
+  configVal :: Maybe Text <-
     liftIO (doesFileExist configPath) >>= \case
-      True -> do
-        mergeWithDefault . toText <$> readFile' configPath
+      True ->
+        Just . toText <$> readFile' configPath
       False ->
-        pure defaultConfig
-  either fail pure $ parsePure configFile $ mergeWithDefault configVal
+        pure Nothing
+  either fail pure $ parseConfig configFile $ mergeWithDefault configVal
 
--- | Pure version of `Dhall.input Dhall.auto`
+-- | Pure version of @'Dhall.input' 'Dhall.auto'@ that also returns the type.
 --
--- The config file cannot have imports, as that requires IO.
-parsePure :: forall a. FromDhall a => FilePath -> Text -> Either String a
+-- The parsed file cannot have imports, as that requires IO.
+parsePure :: FromDhall a => FilePath -> Text -> Either String (a, Dhall.Core.Expr Dhall.Parser.Src Void)
 parsePure fn s = do
   expr0 <- first show $ Dhall.Parser.exprFromText fn s
   expr <- maybeToRight "Cannot have imports" $ traverse (const Nothing) expr0
-  void $ first show $ Dhall.TypeCheck.typeOf expr
-  first show $
-    validationToEither $
-      Dhall.extract @a Dhall.auto $
-        Dhall.Core.normalize expr
+  typ <- first show $ Dhall.TypeCheck.typeOf expr
+  val <- first show $ validationToEither $ Dhall.extract Dhall.auto $ Dhall.Core.normalize expr
+  pure (val, typ)
+
+-- | Parse a Neuron Dhall config file.
+--
+-- The config file cannot have imports, as that requires IO.
+-- The config file also cannot have spurious fields outside of those defined in 'Config'.
+-- (Exception: for backwards compatibility, silently ignore a spurious @mathJaxSupport@ field
+-- if it is the only spurious field.)
+parseConfig :: FilePath -> Text -> Either String Config
+parseConfig fn s = do
+  (config, Dhall.Core.Record (Dhall.Map.keysSet -> fields)) <- parsePure fn s
+  let spuriousFields = fields \\ configKeys
+  if null $ spuriousFields \\ one "mathJaxSupport"
+    then pure config
+    else do
+      let spuriousMessage =
+            "Configuration in neuron.dhall includes spurious fields: " <>
+            intercalate ", " (("`" <>) . (<> "`") . toString <$> toList spuriousFields) <> "."
+          mathJaxMessage =
+            if "mathJaxSupport" `member` spuriousFields
+              then " To disable MathJax support, include a (possibly blank) `./head.html` file."
+              else ""
+      fail $ spuriousMessage <> mathJaxMessage
```

Comment:
Config parsing has gotten a bit complex. I'm wondering if we should just break backwards compact, and have users remove this field manually.